### PR TITLE
Regression Test for Car17

### PR DESCRIPTION
**Pull request by Victor0118**
1. delete the doc for CAR18
2. add the automatically generated doc for CAR17
3. add qrels file and topic file for CAR17-test200
4. update the CarTopicReader to make it work on CAR17

@Peilin-Yang @lintool Could you review this PR?

### CODE REVIEW

File: src/main/resources/regression/car17.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        - 0.2270
+  - name: bm25+rm3
+    params:
+      - -bm25
+      - -rm3
+    results:
+      map:
+        - 0.1354
+      recip_rank:
+        - 0.1861
```
Comment (Peilin-Yang) [inline src/main/resources/regression/car17.yaml]: Why not `BM25+AX` and `QL+AX`?

Comment (Victor0118) [reply src/main/resources/regression/car17.yaml]: Will do it!

**Comment by Victor0118** [comment]
@Peilin-Yang @lintool The axiom results are added. Could you review this PR now?

File: src/main/resources/docgen/templates/car17.template [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
++ `qrel: qrels.car17.test200.hierarchical.txt`: [adhoc qrels (TREC 2017 Complex Answer Retrieval Track)](http://trec-car.cs.unh.edu/datareleases/v1.5/test200-v1.5.tar.xz)
+
+
+After indexing has completed, you should be able to perform retrieval as follows:
+
+```
+${ranking_cmds}
+```
+
+Evaluation can be performed using `trec_eval` and `gdeval.pl`:
```
Comment (Peilin-Yang) [inline src/main/resources/docgen/templates/car17.template]: Seems it is not using `gdeval.pl`

File: src/main/resources/regression/car17.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+search_command: target/appassembler/bin/SearchCollection
+topic_root: src/main/resources/topics-and-qrels/
+qrels_root: src/main/resources/topics-and-qrels/
+index_root:
+ranking_root:
+generator: LuceneDocumentGenerator
+threads: 40
+index_options:
+  - -storePositions
+  - -storeDocvectors
```
Comment (Peilin-Yang) [inline src/main/resources/regression/car17.yaml]: We've decided to enable `-storeRawDocs` for regression test. 
Could you please add it?

File: src/main/resources/regression/car17.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      - -rm3
+    results:
+      map:
+        - 0.1354
+      recip_rank:
+        - 0.1860
+  - name: bm25+ax
+    params:
+      - -bm25
+      - -axiom
```
Comment (Peilin-Yang) [inline src/main/resources/regression/car17.yaml]: In order to make the axiomatic reranking deterministic, you will need to add 
```
- -rerankCutoff 20
- -axiom.deterministic
```
Please see other yaml files for examples

File: src/main/resources/regression/car17.yaml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      - -rm3
+    results:
+      map:
+        - 0.1220
+      recip_rank:
+        - 0.1687
+  - name: ql+ax
+    params:
+      - -bm25
+      - -axiom
```
Comment (Peilin-Yang) [inline src/main/resources/regression/car17.yaml]:
```
- -rerankCutoff 20
- -axiom.deterministic
```

File: src/main/resources/docgen/templates/car17.template [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+The directory `/path/to/Car17` should be the root directory of Car17 collection, i.e., `ls /path/to/Car17` should bring up a list of `.cbor` files.
+
+For additional details, see explanation of [common indexing options](common-indexing-options.md).
+
+## Retrieval
+
+Topics and qrels are stored in `src/main/resources/topics-and-qrels/`, downloaded from NIST:
+
++ `topics.car17.test200.txt`: [Topics for the test200 subset (TREC 2017 Complex Answer Retrieval Track)](http://trec-car.cs.unh.edu/datareleases/v1.5/test200-v1.5.tar.xz)
++ `qrel: qrels.car17.test200.hierarchical.txt`: [adhoc qrels (TREC 2017 Complex Answer Retrieval Track)](http://trec-car.cs.unh.edu/datareleases/v1.5/test200-v1.5.tar.xz)
```
Comment (Peilin-Yang) [inline src/main/resources/docgen/templates/car17.template]: Why is this file called `qrels.car17.test200.hierarchical`, which is not compatible with the topics file?
I think `topics.car17.txt` and `qrels.car17.txt` is better.

Comment (Victor0118) [reply src/main/resources/docgen/templates/car17.template]: In car17, there are two test sets: `test200` and `benchmark_test`. I follow [benchmark paper](https://arxiv.org/abs/1705.04803) and use `test200` as the evaluation set in Anserini.

And for each test set, there are three kinds of `qrels` file: `article`, `toplevel`, `hierarchical`. Car17 use `hierarchical` for the final evaluation.

That's where `test200` and `hierarchical`are from. To clarify which test set and `qrels` file I use,  I add them in the file name. Is that good?

Comment (Peilin-Yang) [reply src/main/resources/docgen/templates/car17.template]: Yea, this makes more sense to me!

File: src/main/resources/regression/car17.yaml [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+index_utils_command: target/appassembler/bin/IndexUtils
+search_command: target/appassembler/bin/SearchCollection
+topic_root: src/main/resources/topics-and-qrels/
+qrels_root: src/main/resources/topics-and-qrels/
+index_root:
+ranking_root:
+generator: LuceneDocumentGenerator
+threads: 40
+index_options:
+  - -storeRawDocs
```
Comment (Peilin-Yang) [inline src/main/resources/regression/car17.yaml]: put `-storeRawDocs` right before `-optimize`

**Pull request merged**